pipeline/test/services/service-cluster/testIngress.sh

```diff
@@ -39,13 +39,18 @@ function check_sc_ingress_health() {
 
     desired_replicas=$(kubectl get daemonset -n ingress-nginx ingress-nginx-controller -ojson | jq ".status.desiredNumberScheduled | tonumber")
     ready_replicas=$(kubectl get daemonset -n ingress-nginx ingress-nginx-controller -ojson | jq ".status.numberReady | tonumber")
+    has_proxy_protocol=$(kubectl get configmap -n ingress-nginx ingress-nginx-controller -oyaml | yq4 '.data.use-proxy-protocol')
 
     diff=$((desired_replicas - ready_replicas))
     if [[ $desired_replicas -eq $ready_replicas ]]; then
         read -r -a pods <<<"$(kubectl get pods -n ingress-nginx -ojson | jq -r '.items[].metadata.name' | tr '\n' ' ')"
         for pod in "${pods[@]}"; do
             if [[ "$pod" =~ ingress-nginx-controller* ]]; then
-                res=$(kubectl -n ingress-nginx exec -it "$pod" -- curl --retry 3 -w "%{http_code}" -o /dev/null -sk https://localhost/healthz)
+                if "${has_proxy_protocol}"; then
+                    res=$(kubectl -n ingress-nginx exec -it "$pod" -- curl --retry 3 --haproxy-protocol -w "%{http_code}" -o /dev/null -sk https://localhost/healthz)
+                else
+                    res=$(kubectl -n ingress-nginx exec -it "$pod" -- curl --retry 3 -w "%{http_code}" -o /dev/null -sk https://localhost/healthz)
+                fi
```

Comment:
Just to minimize on duplicate code, I think I would like to have it something like this:
```suggestion
                extra_flags=""
                if "${has_proxy_protocol}"; then
                    extra_flags+="--haproxy-protocol "
                fi
                res=$(kubectl -n ingress-nginx exec -it "$pod" -- curl --retry 3 ${extra_flags} -w "%{http_code}" -o /dev/null -sk https://localhost/healthz)

```

Reply:
done please check